Assignments/Splitwise/Python/Group.py

```diff
@@ -0,0 +1,109 @@
+from typing import List
+from User import User
+from Expense import Expense
+
+class Group: 
+    def __init__(self, inputList: List[str]): 
+        self.users = {}
+        self.balances = {}
+        self.expenses = []
+
+        usersString = inputList[2:]
+        count = 1
+        for usr in usersString:
+            id = 'u'+str(count)
+            self.users[id] = User(id,usr)
+            count += 1
+
+    def addExpense(self, inputList: List[str]):
+
+        expenseBy = inputList[1]
+        amountToSplit = int(inputList[2])
+        splitAmong = []
+        splitArgs = []
+        expenseType = ""
+
+        amount = amountToSplit
+        num = int(inputList[3])
+        owedBy = inputList[4:4+num]
+
+        for user in owedBy:
+            splitAmong.append(self.users[user])
+
+        splitAmong.append(self.users[expenseBy])
+        splitAmong = set(splitAmong)
+
+        # Update the transaction 
+        def updateAmount(owedTo, usr, amt) :
+            if owedTo == usr : return
+            if (owedTo,usr) in self.balances:
+                self.balances[(owedTo,usr)] += amt
+            elif (usr,owedTo) in self.balances:
+                self.balances[(usr,owedTo)] -= amt
+            else:
+                self.balances[(owedTo,usr)] = amt
+
+
+        if "EQUAL" in inputList:
+            expenseType = "EQUAL"
+            amount = roundNumber(amount / num)
+            for usr in owedBy:
+                updateAmount(expenseBy,usr,amount)
+
+
+        if "EXACT" in inputList:
```

Comment:
Instead of typing this again and again, why dont you create a function which does this. Keep the code as clean as possible

Reply:
I have attempted a refactor, although I think it's not very good.